snapshot-labs/snapshot
# fix: improve validation when uploading image to pineapple

### Issues

Current image upload through pineapple does not validate file size.
Pineapple is returning an error for file > 1MB

We should enforce the file size check on client side to avoid wasting bandwidth, and saving pineapple resources.

Also, there's no error message when user avatar upload fail when updating

Fixes #4109 

### Changes 

1. Add client-side file size validation when uploading image.
2. Error when uploading the user avatar is now displayed in a notification
3. When pineapple.js returns an human readable error, show it instead of the generic "Something went wrong"

### How to test

1. Go to your profile
2. Upload an avatar > 1MB
3. It should display an error, without starting the upload 
4. Rename a non-image file with the .png extension, and upload it
5. It should display a "unsupported file type" error, which is coming directly from pineapple

### Self-review checklist

### Additional notes or considerations
*_(Include any other relevant information or context that may be helpful for the reviewer)_*
